dm/syncer/data_validator.go

```diff
@@ -353,11 +375,11 @@ func (v *DataValidator) fillResult(err error, needLock bool) {
 
 func (v *DataValidator) errorProcessRoutine() {
 	defer v.errProcessWg.Done()
+	// todo: what if this routine panic? we still need to drain v.errChan.
```

Comment:
maybe routineWrapper can be changed to

```
func routineWrapper(workFn func(), recoverFn func(error))
```

in recoverFn we can call v.sendError, v.L, or for range v.errChan

Reply:
> in recoverFn we can call v.sendError, v.L, or for range v.errChan

not easy to recover too. will remove this todo, let it panic, need to check and fix it anyway.